corehq/apps/app_manager/models.py

```diff
@@ -2658,6 +2670,238 @@ def validate_for_build(self):
+        return self._report
+
+    @property
+    def select_detail_id(self):
+        return 'reports.{}.select'.format(self.report_id)
+
+    @property
+    def summary_detail_id(self):
+        return 'reports.{}.summary'.format(self.report_id)
+
+    @property
+    def data_detail_id(self):
+        return 'reports.{}.data'.format(self.report_id)
+
+    def get_details(self):
+        yield (self.select_detail_id, self.select_details(), True)
+        yield (self.summary_detail_id, self.summary_details(), True)
+        yield (self.data_detail_id, self.data_details(), True)
+
+    def select_details(self):
+        return Detail(custom_xml=suite_xml.Detail(
+            id='reports.{}.select'.format(self.report_id),
+            title=suite_xml.Text(
+                locale=suite_xml.Locale(id=id_strings.report_menu()),
+            ),
+            fields=[
+                suite_xml.Field(
+                    header=suite_xml.Header(
+                        text=suite_xml.Text(
+                            locale=suite_xml.Locale(id=id_strings.report_name_header()),
+                        )
+                    ),
+                    template=suite_xml.Template(
+                        text=suite_xml.Text(
+                            xpath=suite_xml.Xpath(function='name'))
+                    ),
+                )
+            ]
+        ).serialize())
+
+    def summary_details(self):
+        def _get_graph_fields():
+            from corehq.apps.userreports.reports.specs import MultibarChartSpec
+            # todo: make this less hard-coded
+            for chart_config in self.report.charts:
+                if isinstance(chart_config, MultibarChartSpec):
+                    def _column_to_series(column):
+                        return suite_xml.Series(
+                            nodeset="instance('reports')/reports/report[@id='{}']/rows/row".format(self.report_id),
+                            x_function='@index',
+                            y_function="column[@id='{}']".format(column),
+                            radius_function='5',
+                        )
+                    yield suite_xml.Field(
+                        header=suite_xml.Header(text=suite_xml.Text()),
+                        template=suite_xml.GraphTemplate(
+                            form='graph',
+                            graph=suite_xml.Graph(
+                                type='xy',
+                                series=[_column_to_series(c) for c in chart_config.y_axis_columns],
+                                configuration=suite_xml.ConfigurationGroup(
+                                    configs=[suite_xml.ConfigurationItem(
+                                        id='x-labels',
+                                        xpath=suite_xml.Xpath(
+                                            function="instance('reports')/reports/report[@id='{}']/xlabels[@column='{}']".format(
```

Comment:
not really sure what to do with this one. i think any change that makes @dimagimon happy makes the code less readable